### PR TITLE
Issue 14846 - Insufficient context deduction with implicit nested lambda

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=14846

Requires Phobos tweak: https://github.com/D-Programming-Language/phobos/pull/3519

### CODE REVIEW

File: src/declaration.c [hunk 1]
```diff
@@ -51,6 +51,8 @@ bool checkFrameAccess(Loc loc, Scope *sc, AggregateDeclaration *ad, size_t iStar
             {
                 if (!fd->isThis() && !fd->isNested())
                     break;
+                if (FuncLiteralDeclaration *fld = fd->isFuncLiteralDeclaration())
+                    fld->tok = TOKdelegate;
```
Comment (WalterBright) [inline src/declaration.c]: `checkFrameAccess` is supposed to be a pure function, merely checking things. But here it is now modifying global state. There's got to be a better way. I do note that `checkNestedReference` does it, too, so there's an excuse for it. I hope that when we switch to ddmd, we can migrate to a better design.

**Comment by WalterBright** [comment]
Auto-merge toggled on

File: src/declaration.c (hunk 1 above)
Comment (9rnsr) [inline src/declaration.c]: Or, the problem is in the function name - `checkSomething` VS `checkAndFixSomething`.

**Comment by 9rnsr** [comment]
Thanks.

**Pull request merged**